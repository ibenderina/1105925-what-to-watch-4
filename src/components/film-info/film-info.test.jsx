import FilmInfo from "@components/film-info/film-info";

it(`Should FilmInfo render correctly`, () => {
  const commonText = `Hello`;
  const commonNumber = 123;

  let tree;
  window.act(() => {
    tree = window.create(<FilmInfo
      film={
        {
          id: commonNumber,
          title: commonText,
          src: commonText,
          description: commonText,
          director: commonText,
          starring: commonText,
          genre: commonText,
          date: commonText,
          background: commonText,
          ratingScore: commonNumber,
          ratingCount: commonNumber,
          url: commonText,
          runTime: commonText,
          comments: [{
            commentAuthor: commonText,
            commentText: commonText,
            commentDate: commonText,
            commentRating: commonNumber,
          }]
        }
      }
    />);
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
