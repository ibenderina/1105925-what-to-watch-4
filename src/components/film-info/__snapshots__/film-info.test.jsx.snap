// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmInfo render correctly 1`] = `
<section
  className="movie-card movie-card--full"
  style={
    Object {
      "backgroundColor": "#CBAC79",
    }
  }
>
  <div
    className="movie-card__hero"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Once Upon a Time in America background"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/ones_upon_a_time_in_america.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link "
          href="/"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          Once Upon a Time in America
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Crime
          </span>
          <span
            className="movie-card__year"
          >
            1984
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <button
            className="btn btn--play movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height={19}
              width={19}
            >
              <use
                style={
                  Object {
                    "fill": undefined,
                  }
                }
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </button>
          <button
            className="btn btn--list movie-card__button"
            onClick={[Function]}
            type="button"
          >
            <svg
              height={20}
              width={19}
            >
              <use
                style={
                  Object {
                    "fill": undefined,
                  }
                }
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
          
        </div>
      </div>
    </div>
  </div>
  <div
    className="movie-card__wrap movie-card__translate-top"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster movie-card__poster--big"
      >
        <img
          alt="Once Upon a Time in America poster"
          height="327"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Once_Upon_a_Time_in_America.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <nav
          className="movie-nav movie-card__nav"
        >
          <ul
            className="movie-nav__list"
          >
            <li
              className="movie-nav__item movie-nav__item--active"
            >
              <a
                className="movie-nav__link"
                href="#"
              >
                Overview
              </a>
            </li>
            <li
              className="movie-nav__item "
            >
              <a
                className="movie-nav__link"
                href="#"
              >
                Details
              </a>
            </li>
            <li
              className="movie-nav__item "
            >
              <a
                className="movie-nav__link"
                href="/films/1/review"
                onClick={[Function]}
              >
                Reviews
              </a>
            </li>
          </ul>
        </nav>
        test
      </div>
    </div>
  </div>
</section>
`;
